src/battle/trainer_ai.c

```diff
@@ -0,0 +1,64 @@
+#include "battle/battle.h"
+#include "battle/battle_system.h"
+#include "battle/trainer_ai.h"
+#include "battle/overlay_12_0224E4FC.h"
+
+void ov10_0221BE20(BattleSystem *bsys, BattleContext *ctx, u8 battlerId, u8 a3) {
+    int i;
+    u8 struggleFlags;
+    u8 *aiData = (u8 *) &ctx->trainerAIData;
+    
+    for (i = 0; i < (u32) 28; i++) {
```

Comment:
is the cast to u32 strictly necessary here? if it's an unsigned comparison couldn't i just be a u32? failing that 28u would be preferable

Reply:
iirc it was an unsigned comparison with i later being explicitly referred to as an int

Reply:
This might be a `memset` call. See also my comment on include/battle/battle.h#L111

Reply:
Checksum failed when attempting to use memset